### DISCUSSION TITLE
symbiotic google drive colab goodness

### DISCUSSION THREAD

**Discussion by alamonelf** [comfyanonymous/ComfyUI#242] [General]
TY ILY COMFY is EPIC.
I would like to get comfy to use my google drive model folder in colab please.

**Comment by WASasquatch** [comment]
Add to custom paths: https://github.com/comfyanonymous/ComfyUI/blob/master/extra_model_paths.yaml.example

There is the snippet you can add, which adds google drive (from one of the menus) just add to a new cell, run it, then add the path to models folders or others to custom paths.

**Comment by alamonelf** [comment]
This is what worked for me.

On google drive I created a text file called extra_model_paths.yaml 
put this in it
______________________________________________________________________

#Rename this to extra_model_paths.yaml and ComfyUI will load it

#config for a1111 ui
#all you have to do is change the base_path to where yours is installed
a111:
    base_path: /content/drive/MyDrive

    checkpoints: /content/drive/MyDrive/models/checkpoints
    configs: /content/drive/MyDrivemodels/configs
    vae: /content/drive/MyDrivemodels/vae
    loras: /content/drive/MyDrivemodels/Lora
    upscale_models: |
                  /content/drive/MyDrivemodels/upscale_models
                  models/SwinIR
    embeddings: /content/drive/MyDrive/models/embeddings
    controlnet: /content/drive/MyDrive/models/ControlNet


#other_ui:
#    base_path: path/to/ui
#    checkpoints: models/checkpoints
___________________________________________________________________

In colab add a cell
________________________________

from google.colab import drive
drive.mount('/content/drive')

_________________________________

Then in another cell

_____________________________________________________________

!gdown --id (google hash for extra_model_paths.yaml file)

_______________________________________________________________

Thank you Comfyui creators and lovers.

**Reply by WASasquatch** [reply]
> This is what worked for me.
> 
> On google drive I created a text file called extra_model_paths.yaml put this in it
> 
> #Rename this to extra_model_paths.yaml and ComfyUI will load it
> 
> #config for a1111 ui #all you have to do is change the base_path to where yours is installed a111: base_path: /content/drive/MyDrive
> 
> ```
> checkpoints: /content/drive/MyDrive/models/checkpoints
> configs: /content/drive/MyDrivemodels/configs
> vae: /content/drive/MyDrivemodels/vae
> loras: /content/drive/MyDrivemodels/Lora
> upscale_models: |
>               /content/drive/MyDrivemodels/upscale_models
>               models/SwinIR
> embeddings: /content/drive/MyDrive/models/embeddings
> controlnet: /content/drive/MyDrive/models/ControlNet
> ```
> 
> #other_ui:
> # base_path: path/to/ui
> # checkpoints: models/checkpoints
> 
> In colab add a cell
> 
> from google.colab import drive drive.mount('/content/drive')
> 
> Then in another cell
> 
> !gdown --id (google hash for extra_model_paths.yaml file)
> 
> Thank you Comfyui creators and lovers.

You could also install ComfyUI on Google Drive itself, and not worry about any file stuff, just use it as normal, running from GDrive folder like many of the Stable Diffusion notebooks that run of your gdrive to save states/files/settings. 

To do so you'd just need to login to gdrive, and cd into the folder and then proceed to run install scripts.
